src/code42cli/cmds/securitydata.py

```diff
@@ -358,23 +365,85 @@ def search(
     include_all,
     **kwargs,
 ):
+
     """Search for file events."""
-    output_header = ext.try_get_default_header(
+    output_header = try_get_default_header(
         include_all, _create_search_header_map(), format
     )
     formatter = FileEventsOutputFormatter(format, output_header)
     cursor = _get_cursor(state, use_checkpoint)
-    handlers = ext.create_handlers(
-        state.sdk,
-        FileEventExtractor,
-        cursor,
-        use_checkpoint,
-        formatter=formatter,
-        force_pager=include_all,
-    )
-    _extract(
-        state, handlers, begin, end, or_query, advanced_query, saved_search, **kwargs
-    )
+    if use_checkpoint:
+        checkpoint_name = use_checkpoint
+        # if checkpoint name exists, checkpoint should be that eventId,
+        # otherwise it should set the initial value to ""
+        checkpoint = cursor.get(checkpoint_name) or ""
+        # older app versions stored checkpoint as float timestamp.
+        # we handle those here until the next run containing events will store checkpoint as the last eventId
+        if isinstance(checkpoint, (int, float)):
+            state.search_filters.append(InsertionTimestamp.on_or_after(checkpoint))
+            checkpoint = ""
+    else:
+        checkpoint = ""
+
+    query = _construct_query(state, begin, end, saved_search, advanced_query, or_query)
+    events = _get_all_file_events(state, query, checkpoint)
+
+    if not events:
+        click.echo("No results found.")
+        return
+
+    if use_checkpoint:
+        checkpoint_name = use_checkpoint
+        # update checkpoint to eventId of last event retrieved
+        events_gen = _store_updated_checkpoint(cursor, checkpoint_name, events)
+        formatter.echo_formatted_generated_output(events_gen)
+    else:
+        formatter.echo_formatted_list(events)
+
+
+def _construct_query(state, begin, end, saved_search, advanced_query, or_query):
+
+    if advanced_query:
+        state.search_filters = advanced_query
+    if saved_search:
+        state.search_filters = saved_search._filter_group_list
+    else:
+        if begin or end:
+            state.search_filters.append(
+                create_time_range_filter(f.EventTimestamp, begin, end)
+            )
+    if or_query:
+        state.search_filters = convert_to_or_query(state.search_filters)
+    verify_filter_groups(state.search_filters)
+    query = FileEventQuery(*state.search_filters)
+    query.page_size = MAX_EVENT_PAGE_SIZE
+    query.sort_direction = "asc"
+    query.sort_key = "insertionTimestamp"
+    return query
+
+
+def _get_all_file_events(state, query, checkpoint=""):
```

Comment:
Because of the page tokens, would it be better for this to return a generator of either events or responses and then for it to handle each individually and update the checkpoint as it goes, rather than building the whole list right away?

Reply:
Yeah, I would lean towards just yielding each event individually from `_get_all_file_events()`, then when using a checkpoint you just wrap the event generator with `_store_updated_checkpoint()` and pass it to `echo_formatted_generated_output()` or pass it there directly.